mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -74,12 +74,26 @@ def stop_app(self):
 
     def _is_app_installed(self):
         """Overrides superclass."""
+        # Check if app is installed
+        try:
+            out = self._adb.shell(
+                'pm list package | grep %s' % self.package).decode('utf-8')
+            if not bool(out):
+                return False
+        except adb.AdbError as e:
+            if (e.ret_code == 1) and (not e.stdout) and (not e.stderr):
+                return False
+            raise
+        # If app is installed, is it listed as an instrumented package.
         try:
             out = self._adb.shell(
                 'pm list instrumentation | grep ^instrumentation:%s/' %
                 self.package).decode('utf-8')
-            return bool(out)
+            is_instrumented = bool(out)
         except adb.AdbError as e:
             if (e.ret_code == 1) and (not e.stdout) and (not e.stderr):
-                return False
+                is_instrumented = False
             raise
+        if not is_instrumented:
```

Comment:
Er wait, no, I'm asking you to remove is_app_installed.

Don't quite understand your comment then. You want to change is_app_running to check_app_running and in the retry loop catch and handle the exception instead of handling boolean output?